forest/src/daemon.rs

```diff
@@ -123,12 +123,17 @@ pub(super) async fn start(config: Config) {
     // Block until ctrl-c is hit
     block_until_sigint().await;
 
+    let keystore_write = task::spawn(async move {
+        keystore.read().await.flush().unwrap();
+    });
+
```

Comment:
Is there a reason why keystore_write.await isn't put directly below this chunk of code?
```suggestion
    let keystore_write = task::spawn(async move {
        keystore.read().await.flush().unwrap();
    });
    keystore_write.await;
```

Reply:
because I don't want the main thread to wait to cancel all tasks before awaiting on the keystore to be written, but I also don't want to wait to start the write until after the tasks are cancelled